skyportal/handlers/api/weather.py

```diff
@@ -60,11 +68,19 @@ def get(self):
                               type: string
                               description: Weather fetching error message
         """
-        user_prefs = getattr(self.current_user, 'preferences', None) or {}
+        user = (
+            User.query_records_accessible_by(self.current_user)
+            .filter(User.username == self.associated_user_object.username)
+            .first()
+        )
+        user_prefs = getattr(user, 'preferences', None) or {}
         weather_prefs = user_prefs.get('weather', {})
         weather_prefs = {**default_prefs, **weather_prefs}
 
-        telescope_id = int(weather_prefs["telescopeID"])
+        default_telescope_id = int(weather_prefs["telescopeID"])
```

Comment:
Are we guaranteed that weather_prefs["telescopeID"] is set? Just checking.

Reply:
I also had this question